pypeit/reduce.py

```diff
@@ -412,13 +424,17 @@ def run(self, basename=None, ra=None, dec=None, obstime=None,
         # Finish up
         if self.sobjs.nobj == 0:
             msgs.warn('No objects to extract!')
-        else:
+        elif self.par['flexure']['spec_method'] not in ['skip', 'slit_cen']:
+            # Apply a refined estimate of the flexure to objects, and then apply reference frame correction to objects
             # TODO -- Should we move these to redux.run()?
             # Flexure correction if this is not a standard star
-            if not self.std_redux:
-                self.spec_flexure_correct(self.sobjs, basename)
-            # Heliocentric
-            self.helio_correct(self.sobjs, ra, dec, obstime)
+            if (not self.std_redux):
+                self.spec_flexure_correct(basename, mode='object', specObjs=self.sobjs, update_waveimg=False)
+            # Apply a reference frame correction to each object
+            self.refframe_correct(ra, dec, obstime, specObjs=self.sobjs, update_waveimg=False)
+
```

Comment:
Can you just consolidate these two calls to reframe_correct into a single call to reframe_correct, that simply decides what to do based on the self.par['flexure']['spec_method']. That would simplify the logic in self.run and 
then we would not need these update_waveimg=True, False toggles, i.e. you can just do the right thing based on the self.par['flexure']['spec_method']

On that note, shouldn't the there be a separate parameter for reframe correction, i.e. we are currently assuming the user wants both objects extracted and the 2-d waveimg hello centric corrected.

Reply:
Okay I think I agree. For some reason I thought a user might want the wavelength image to be uncorrected but the objects corrected, but I doubt that is a desired usage case.